### ISSUE TITLE
 Can't execute: Malformed tests.yml when environment is present

### ISSUE THREAD

**Issue by lukaszachy** [psss/tmt#126]
I am not able to execute test if it  sets the environment.

Steps to reproduce:
Plan in file repro.fmf with content
```yaml
summary:
    Basic smoke test
discover:
    how: fmf
execute:
    how: beakerlib
```
Test in main.fmf with content
```yaml
test: ./runtest.sh
environment: {'BUNDLED_INJECT': 'PYTHONPATH=/usr/lib/python3.6/site-packages/pip/_vendor/',
 'PACKAGES': 'python3-pip',
 'PYTHON': 'python3'}
```

Execution as ```tmt run -d``` ends up with
``` 
   execute
        workdir: /var/tmp/tmt/run-055/repro/execute
        how: beakerlib
        Copy '/usr/lib/python3.7/site-packages/tmt/steps/execute/run.sh' to '/var/tmp/tmt/run-055/repro/execute'.
            Run command 'vagrant rsync'.
            out: ==> default: Rsyncing folder: /var/tmp/tmt/run-055/repro/ => /var/tmp/tmt/run-055/repro
            Run command 'vagrant ssh -c nohup /var/tmp/tmt/run-055/repro/execute/run.sh -v /var/tmp/tmt/run-055/repro beakerlib /var/tmp/tmt/run-055/repro/execute/stdout.log /var/tmp/tmt/run-055/repro/execute/stderr.log'.
            out: nohup: ignoring input and appending output to 'nohup.out'
            err: Connection to 192.168.122.226 closed.
            Run command 'bash -c "vagrant plugin list | grep '^vagrant-rsync-back '"'.
            out: vagrant-rsync-back (0.0.1, system)
            Run command 'vagrant rsync-back'.
            out: ==> default: Rsyncing folder: /var/tmp/tmt/run-055/repro/ => /var/tmp/tmt/run-055/repro
    stdout.log: ED

    stderr.log: 
         /var/tmp/tmt/run-055/repro $ main beakerlib < discover/tests.yaml
         > /one/repro:
         >     environment: BUNDLED_INJECT=PYTHONPATH=/usr/lib/python3.6/site-packages/pip/_vendor/
         >         PACKAGES=python3-pip PYTHON=python3
        Error: Unknown test variable:         PACKAGES=python3-pip PYTHON=python3
         >     path: /one/tests/repro
         >     test: ./runtest.sh
        Error: [/one/repro] Could not find test dir: '/var/tmp/tmt/run-055/repro/discover/one/tests/repro'
        overview: E
        result: 0 tests passed, 0 tests failed
 ERROR  1 errors occured during tests.
```

tests.yml from discover step has content
```yaml
/one/repro:
    environment: BUNDLED_INJECT=PYTHONPATH=/usr/lib/python3.6/site-packages/pip/_vendor/
        PACKAGES=python3-pip PYTHON=python3
    path: /one/tests/repro
    test: ./runtest.sh
```

**Comment by pvalena** [comment]
```
        Error: Unknown test variable:         PACKAGES=python3-pip PYTHON=python3
```
What this says is that `run.sh` can't  parse it (it's read from `tests.yaml` file). It is caused by the simple `yaml-like` format data parser (it's not full yaml-reader), and I'd rather avoid implementing it in bash, if possible.

What I'd recommend is to have the long line not-wrapped, which would simply make it work again. @psss, can we fix that on the generator / python side?

**Comment by pvalena** [comment]
```
        Error: [/one/repro] Could not find test dir: '/var/tmp/tmt/run-055/repro/discover/one/tests/repro'
```

@lukaszachy is the dir intentionally nonexistent?

**Comment by lukaszachy** [comment]
> ```
>         Error: [/one/repro] Could not find test dir: '/var/tmp/tmt/run-055/repro/discover/one/tests/repro'
> ```
> 
> @lukaszachy is the dir intentionally nonexistent?

Yes, I was trying to create simpler reproducer.

**Comment by psss** [comment]
Should be fixed by #188.
